### PR TITLE
Passenger/Platform API | Dynamic passenger/ride options

### PR DESCRIPTION
**Pull request by phylor**
This adds the options for the dynamic passenger/ride options to the passenger and platform API.

### CODE REVIEW

File: lib/ioki/model/platform/ride_option.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Ioki
+  module Model
+    module Platform
+      class RideOption < Base
+        attribute :name, on: [:read, :create], type: :string
+        attribute :value, on: :create, type: :string # could also be a boolean or an integer
```
Comment (rmehner) [inline lib/ioki/model/platform/ride_option.rb:8]: I briefly looked into supporting more that one type and it's a little bit more involved. Just feels weird that we might pass a boolean here and have it sent as string. Can the API handle that?

We'd need to adapt [type_cast_attribute_value](https://github.com/ioki-mobility/ioki-ruby/blob/dffab1bdfe37e136ce9728dfd8075d6ab8ed69bb/lib/ioki/model/base.rb#L124-L167) to work with an array of possible types and then check the incoming value against those types. 

Is that worth it?

**Review by rmehner — COMMENTED** [review]
Generally looks good, just missing some specs :)

File: lib/ioki/model/platform/ride_option.rb (hunk 1 above)
Comment (phylor) [reply lib/ioki/model/platform/ride_option.rb]: > I briefly looked into supporting more that one type and it's a little bit more involved. Just feels weird that we might pass a boolean here and have it sent as string. Can the API handle that?

No, I don't think so. The API is quite picky about the right data types 😀 I only tested with strings so far.

I added support for it in b9f9212e75a4051dbd5cd8347f0ac687f31828dd.

**Comment by phylor** [comment]
> Generally looks good, just missing some specs :)

I added specs for the multi-type attributes. Would you expect other specs?

**Review by tom-ioki — APPROVED** [review]
Looks good to me and sounds like a reasonable idea 👍

**Comment by tom-ioki** [comment]
Thanks for the detailed information 🙏

**Comment by rmehner** [comment]
> If there is no matched type, we return `nil`. We could also return the parsed value (although the type doesn't match) or raise an exception.

I'm wondering what you'd expect here as a user of the client.

1. Exception might be way too harsh, because if Triebwerk ever changes something, we don't want clients to fail hard. Although it would help getting things fixed quickly 😄 
2. Returning `nil` would mean that the value is lost and it might lead to a subtle bug if someone expects something else. It would allow for proper nil checking on consumer side though, which might be a good pattern to have in consuming apps.
3. Returning the value parsed as is seems to be the best variant to me, as this would allow consumers to work with those values, even if ioki-ruby wasn't updated yet. Downside is that this doesn't put pressure on ioki-ruby to be up to date and we devs love weird workarounds :0

So I'm slightly in favor of 3, but 2 would also be okay with me.

**Comment by phylor** [comment]
> 2. Returning `nil` would mean that the value is lost and it might lead to a subtle bug if someone expects something else. It would allow for proper nil checking on consumer side though, which might be a good pattern to have in consuming apps.
> 3. Returning the value parsed as is seems to be the best variant to me, as this would allow consumers to work with those values, even if ioki-ruby wasn't updated yet. Downside is that this doesn't put pressure on ioki-ruby to be up to date and we devs love weird workarounds :0
> 
> So I'm slightly in favor of 3, but 2 would also be okay with me.

Hmm, I'm a bit indifferent here :slightly_smiling_face: I generally prefer "fail early", so I'd prefer to return `nil` instead of a wrong type.

I'll go with the path of least resistance and merge as is (with option 2).

@rmehner What do you think: we change it to option 3 (returning whatever value we got) and then we additionally do

```ruby
STDERR.puts "We expected `Ioki::Model::Passenger::LineStop::supports_pickup` to be a `boolean`, but got 'i_am_a_string'"
```

**Pull request merged**